File: output/ChatGPT/avatar/Python/C++/codeforces_644_B.cpp
#include <iostream>
#include <queue>
#include <map>

using namespace std;

struct Task {
    int time;
    int duration;
    int index;
};

int main() {
    int n, b;
    cin >> n >> b;

    queue<Task> remaining;
    bool running = false;
    int finish_time = 0;
    map<int, int> result;

    for (int i = 0; i < n; i++) {
        int time, duration;
        cin >> time >> duration;

        Task task;
        task.time = time;
        task.duration = duration;
        task.index = i;

        result[i] = 0;

        if (task.time > finish_time && remaining.empty()) {
            running = true;
            finish_time = task.time + task.duration;
            result[i] = finish_time;
        } else {
            if (task.time >= finish_time && !remaining.empty()) {
                // Run task
                Task task_to_run = remaining.front();
                remaining.pop();
                finish_time = max(finish_time, task_to_run.time) + task_to_run.duration;
                result[task_to_run.index] = finish_time;
            }

            if (remaining.size() < b) {
                remaining.push(task);
            } else {
                result[i] = -1;
            }
        }
    }

    while (!remaining.empty()) {
        // Run task
        Task task_to_run = remaining.front();
        remaining.pop();
        finish_time = max(finish_time, task_to_run.time) + task_to_run.duration;
        result[task_to_run.index] = finish_time;
    }

    for (auto const& pair : result) {
        cout << pair.second << endl;
    }

    return 0;
}

File: output/ChatGPT/avatar/Python/Java/codeforces_644_B.java
import java.util.*;

class Task {
    int time;
    int duration;
    int index;

    public Task(int time, int duration, int index) {
        this.time = time;
        this.duration = duration;
        this.index = index;
    }
}

public class codeforces_644_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int b = scanner.nextInt();

        Queue<Task> remaining = new LinkedList<>();
        boolean running = false;
        int finishTime = 0;

        Map<Integer, Integer> result = new HashMap<>();

        for (int i = 0; i < n; i++) {
            int time = scanner.nextInt();
            int duration = scanner.nextInt();

            Task task = new Task(time, duration, i);
            result.put(i, 0);

            if (task.time > finishTime && remaining.isEmpty()) {
                running = true;
                finishTime = task.time + task.duration;
                result.put(i, finishTime);
            } else {
                if (task.time >= finishTime && !remaining.isEmpty()) {
                    finishTime = runTask(remaining, finishTime, result);
                }
                if (remaining.size() < b) {
                    remaining.add(task);
                } else {
                    result.put(i, -1);
                }
            }
        }

        while (!remaining.isEmpty()) {
            finishTime = runTask(remaining, finishTime, result);
        }

        for (int key : result.keySet()) {
            System.out.println(result.get(key));
        }
    }

    public static int runTask(Queue<Task> remaining, int finishTime, Map<Integer, Integer> result) {
        Task taskToRun = remaining.poll();
        finishTime = Math.max(finishTime, taskToRun.time) + taskToRun.duration;
        result.put(taskToRun.index, finishTime);
        return finishTime;
    }
}